opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/export/__init__.py

```diff
@@ -137,20 +143,66 @@ class BatchLogRecordProcessor(LogRecordProcessor):
     """This is an implementation of LogRecordProcessor which creates batches of
     received logs in the export-friendly LogData representation and
     send to the configured LogExporter, as soon as they are emitted.
+
+    `BatchLogRecordProcessor` is configurable with the following environment
+    variables which correspond to constructor parameters:
+
+    - :envvar:`OTEL_BLRP_SCHEDULE_DELAY`
+    - :envvar:`OTEL_BLRP_MAX_QUEUE_SIZE`
+    - :envvar:`OTEL_BLRP_MAX_EXPORT_BATCH_SIZE`
+    - :envvar:`OTEL_BLRP_EXPORT_TIMEOUT`
     """
 
     def __init__(
         self,
         exporter: LogExporter,
-        schedule_delay_millis: int = 5000,
-        max_export_batch_size: int = 512,
-        export_timeout_millis: int = 30000,
+        max_queue_size: int = None,
+        schedule_delay_millis: int = None,
+        max_export_batch_size: int = None,
+        export_timeout_millis: int = None,
     ):
+        if max_queue_size is None:
+            max_queue_size = int(environ.get(OTEL_BLRP_MAX_QUEUE_SIZE, 2048))
+
+        if schedule_delay_millis is None:
+            schedule_delay_millis = int(
+                environ.get(OTEL_BLRP_SCHEDULE_DELAY, 5000)
+            )
+
+        if max_export_batch_size is None:
+            max_export_batch_size = int(
+                environ.get(OTEL_BLRP_MAX_EXPORT_BATCH_SIZE, 512)
+            )
+
+        if export_timeout_millis is None:
+            export_timeout_millis = int(
+                environ.get(OTEL_BLRP_EXPORT_TIMEOUT, 30000)
+            )
```

Comment:
All of this env should handle non-number values for `int()`. 

```python
>>> from os import environ
>>> environ['OTEL_BLRP_EXPORT_TIMEOUT'] = 'foo'
>>> int(environ.get('OTEL_BLRP_EXPORT_TIMEOUT', 30000))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'foo'
```

Reply:
Done. Added ValueError handling and tests for BLRP as well as BSP.